app/models/pglogical_subscription.rb

```diff
@@ -178,6 +190,15 @@ def self.find_id(to_find)
 
   private
 
+  def safe_delete
+    pglogical.drop_subscription(id, true)
+  rescue PG::InternalError => e
+    raise unless e.message =~ /could not connect to publisher/ || e.message =~ /replication slot .* does not exist/
+    disable
+    pglogical.alter_subscription_options(id, "slot_name" => "NONE")
+    pglogical.drop_subscription(id, true)
```

Comment:
What happens if these blow up in the rescue clause here...could we be left in a weird state?

Reply:
Maybe? Considering we're explicitly not contacting the remote server in this case maybe we can do these three in a transaction, I'll give it a shot.